Src/GoogleApis.Auth/OAuth2/ClientCredentialParameters.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+Copyright 2015 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Google.Apis.Json;
+using Google.Apis.Util;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>Indicates the type of Client Credential.</summary>
+    public enum ClientCredentialType
+    {
+        /// <summary>UserCredential is created by the gcloud sdk tool when the user runs <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        UserCredential = 1,
+        /// <summary>ServiceAccountCredential is downloaded by the user from <a href="https://console.developers.google.com">Google Developers Console</a>.</summary>
+        ServiceAccountCredential = 2
+    }
+
+    /// <summary>Container for the union of parameters/properties from all credential types.</summary>
+    public sealed class ClientCredentialParameters
```

Comment:
So, to explain this a bit more:

there's JSON file at well known location "application_default_credentials.json", which contains either the params for service account credentials   OR params for the user credential.

The point is that your app does't need to care if you are running in GCE or using a service account on a server or if is just on your localhost and you want to use you personal user credentials.

So the app always goes to the well known file (unless we're on GCE) and sees if it contains user credentials (for which you've already given consent using gcloud auth login) or service account credentials and instantiates the right class for you.

In this PR, the JSON data object (now called ClientCredentialParameters) is only used for ServiceAccountCredential. But in the followup PR it might contain UserCredential as well.

So after looking at the class and its summary again, the name ClientCredentialParameters makes more sense to me.